collections/src/main/scala/strawman/collection/mutable/Stack.scala

```diff
@@ -38,46 +39,63 @@ class Stack[A] protected (array: Array[AnyRef], start: Int, end: Int)
     * @param elem
     * @return
     */
-  def push(elem: A): this.type = this += elem
+  def push(elem: A): this.type = prepend(elem)
 
   /** Push two or more elements onto the stack. The last element
     *  of the sequence will be on top of the new stack.
     *
     *  @param   elems      the element sequence.
     *  @return the stack with the new elements on top.
     */
-  def push(elem1: A, elem2: A, elems: A*): this.type = push(elem1).push(elem2).pushAll(elems.toStrawman)
+  def push(elem1: A, elem2: A, elems: A*): this.type = {
+    val k = elems.toStrawman.knownSize
+    ensureSize(length + (if(k >= 0) k + 2 else 3))
+    prepend(elem1).prepend(elem2).pushAll(elems.toStrawman)
+  }
 
   /** Push all elements in the given traversable object onto the stack. The
     *  last element in the traversable object will be on top of the new stack.
     *
     *  @param elems the traversable object.
     *  @return the stack with the new elements on top.
     */
-  def pushAll(elems: strawman.collection.IterableOnce[A]): this.type = this ++= elems
+  def pushAll(elems: strawman.collection.IterableOnce[A]): this.type =
+    prependAll(elems match {
+      case it: strawman.collection.Seq[A] => it.view.reverse
+      case it => IndexedSeq.from(it).view.reverse
+    })
```

Comment:
We see here that it’s less efficient to choose this order (pushed elements are prepended rather than appended). Is it specified somewhere that it should be done that way?

Reply:
It's not less efficient in general but the implementation is not ideal. Note that `prependAll` converts to an `IndexedSeq` anyway if the size is not known, so the amount of copying shouldn't change, it's only an extra layer of indirection through the view.